src/80211.c

```diff
@@ -132,13 +132,19 @@ int8_t signal_strength(const u_char *packet, size_t len)
 	int8_t ssi = 0;
 	int offset = sizeof(struct radio_tap_header);
 	struct radio_tap_header *header = NULL;
+	uint32_t flags, flags2;
 
 	if(has_rt_header() && (len > (sizeof(struct radio_tap_header) + TSFT_SIZE + FLAGS_SIZE + RATE_SIZE + CHANNEL_SIZE + FHSS_FLAG)))
 	{
 		header = (struct radio_tap_header *) packet;
 
-		uint32_t flags = __le32_to_cpu(header->flags);
-		
+		flags = flags2 = __le32_to_cpu(header->flags);
+		while ((flags2 & (1u << 31)) && offset <= len - 4)
+		{
+			flags2 = __le32_to_cpu(*(uint32_t *)(packet + offset));
+			offset += sizeof(flags2);
+		}
```

Comment:
interesting. we were just discussing this issue, that the EXT bit might be the cause of the dbm-zero issue (#69), since i encountered [the exact same issue in my macgeiger project](https://github.com/rofl0r/MacGeiger/commit/ac71d305a272d6d5ec2ff9d4922c906f49025ed9). however in my case i would have to [account for the padding](https://github.com/radiotap/radiotap-library/commit/9cb10453c50681a328cfea130a7e82ca3d682e05#commitcomment-20973654). won't this be needed here too?